File: lib/presentation/views/shared/sunday_dedication_view.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../../models/habit.dart';
import '../../models/scripture.dart';
import '../../providers/habit_provider.dart';
import '../../services/week_cycle_manager.dart';
import '../../theme/app_theme.dart';

class SundayDedicationView extends StatefulWidget {
  final WeekCycleManager weekCycleManager;
  final VoidCallback onDismiss;

  const SundayDedicationView({
    super.key,
    required this.weekCycleManager,
    required this.onDismiss,
  });

  @override
  State<SundayDedicationView> createState() => _SundayDedicationViewState();
}

class _SundayDedicationViewState extends State<SundayDedicationView>
    with SingleTickerProviderStateMixin {
  bool _showHeading = false;
  bool _showVerse = false;
  bool _showButton = false;
  bool _isDedicating = false;
  bool _isDedicated = false;
  bool _showDedicatedMessage = false;
  double _glowIntensity = 0.08;
  int _revealedTileCount = 0;

  late AnimationController _breatheController;
  late Animation<double> _breatheAnim;

  @override
  void initState() {
    super.initState();
    _breatheController = AnimationController(
      vsync: this,
      duration: const Duration(seconds: 3),
    )..repeat(reverse: true);
    _breatheAnim = Tween(begin: 0.0, end: 1.0).animate(_breatheController);
    _startEntryAnimations();
  }

  @override
  void dispose() {
    _breatheController.dispose();
    super.dispose();
  }

  void _startEntryAnimations() {
    Future.delayed(const Duration(milliseconds: 300), () {
      if (mounted) setState(() => _showHeading = true);
    });

    final habits = context.read<HabitProvider>().sortedHabits;
    for (int i = 0; i <= habits.length; i++) {
      Future.delayed(Duration(milliseconds: 800 + i * 200), () {
        if (mounted) setState(() => _revealedTileCount = i + 1);
      });
    }

    final verseDelay = 800 + habits.length * 200 + 300;
    Future.delayed(Duration(milliseconds: verseDelay), () {
      if (mounted) setState(() => _showVerse = true);
    });
    Future.delayed(Duration(milliseconds: verseDelay + 300), () {
      if (mounted) setState(() => _showButton = true);
    });
  }

  Future<void> _performDedication() async {
    setState(() {
      _isDedicating = true;
      _glowIntensity = 0.3;
    });
    await Future.delayed(const Duration(milliseconds: 1000));
    if (!mounted) return;
    await widget.weekCycleManager.dedicateCurrentWeek();
    setState(() {
      _isDedicated = true;
      _showButton = false;
      _glowIntensity = 0.12;
    });
    await Future.delayed(const Duration(milliseconds: 500));
    if (mounted) setState(() => _showDedicatedMessage = true);
  }

  bool get _isMidWeekStart =>
      !widget.weekCycleManager.isSunday;

  @override
  Widget build(BuildContext context) {
    final habits = context.watch<HabitProvider>().sortedHabits;

    return Scaffold(
      backgroundColor: TributeColor.charcoal,
      body: Stack(
        children: [
          // Animated glow background
          AnimatedBuilder(
            animation: _breatheAnim,
            builder: (_, child) => Container(
              decoration: BoxDecoration(
                gradient: RadialGradient(
                  colors: [
                    TributeColor.golden.withValues(alpha: _glowIntensity),
                    TributeColor.softGold
                        .withValues(alpha: _glowIntensity * 0.4),
                    Colors.transparent,
                  ],
                  radius: 0.7 + _breatheAnim.value * 0.1,
                ),
              ),
            ),
          ),
          SafeArea(
            child: Column(
              children: [
                Expanded(
                  child: SingleChildScrollView(
                    padding: const EdgeInsets.fromLTRB(24, 32, 24, 100),
                    child: _isDedicated
                        ? _postDedicationContent()
                        : _preDedicationContent(habits.toList()),
                  ),
                ),
                if (!_isDedicated && _showButton && !_isDedicating)
                  Padding(
                    padding: const EdgeInsets.fromLTRB(24, 0, 24, 32),
                    child: SizedBox(
                      width: double.infinity,
                      child: ElevatedButton.icon(
                        onPressed: _performDedication,
                        icon: const Icon(Icons.auto_awesome, size: 18),
                        label: const Text('Dedicate this week to God',
                            style: TextStyle(fontWeight: FontWeight.w600, fontSize: 16)),
                        style: ElevatedButton.styleFrom(
                          backgroundColor: TributeColor.golden,
                          foregroundColor: TributeColor.charcoal,
                          padding: const EdgeInsets.symmetric(vertical: 16),
                          shape: RoundedRectangleBorder(
                              borderRadius: BorderRadius.circular(14)),
                        ),
                      ),
                    ),
                  ),
                if (_isDedicated && _showDedicatedMessage)
                  Padding(
                    padding: const EdgeInsets.fromLTRB(24, 0, 24, 32),
                    child: SizedBox(
                      width: double.infinity,
                      child: ElevatedButton.icon(
                        onPressed: widget.onDismiss,
                        icon: const Icon(Icons.arrow_forward, size: 18),
                        label: const Text('Begin your week',
                            style: TextStyle(fontWeight: FontWeight.w600, fontSize: 16)),
                        style: ElevatedButton.styleFrom(
                          backgroundColor: TributeColor.golden,
                          foregroundColor: TributeColor.charcoal,
                          padding: const EdgeInsets.symmetric(vertical: 16),
                          shape: RoundedRectangleBorder(
                              borderRadius: BorderRadius.circular(14)),
                        ),
                      ),
                    ),
                  ),
              ],
            ),
          ),
        ],
      ),
    );
  }

  Widget _preDedicationContent(List<Habit> habits) {
    return Column(
      children: [
        AnimatedOpacity(
          opacity: _showHeading ? 1 : 0,
          duration: const Duration(milliseconds: 500),
          child: AnimatedSlide(
            offset: _showHeading ? Offset.zero : const Offset(0, 0.2),
            duration: const Duration(milliseconds: 500),
            child: Column(
              children: [
                if (_isMidWeekStart)
                  Text('Starting mid-week?',
                      textAlign: TextAlign.center,
                      style: TextStyle(
                          fontSize: 12,
                          color: TributeColor.softGold.withValues(alpha: 0.6))),
                const SizedBox(height: 6),
                Text(
                  _isMidWeekStart
                      ? 'No problem. Let\u2019s dedicate\nwhat\u2019s left of this week.'
                      : 'Set Your Week',
                  textAlign: TextAlign.center,
                  style: const TextStyle(
                      fontSize: 22,
                      fontWeight: FontWeight.w700,
                      color: TributeColor.warmWhite),
                ),
                const SizedBox(height: 8),
                Text(
                  'Your habits, your purpose, your offering.',
                  textAlign: TextAlign.center,
                  style: TextStyle(fontSize: 14, color: Colors.white.withValues(alpha: 0.5)),
                ),
              ],
            ),
          ),
        ),
        const SizedBox(height: 28),
        ...habits.asMap().entries.map((e) {
          final i = e.key;
          final habit = e.value;
          final isRevealed = i < _revealedTileCount;
          return AnimatedOpacity(
            opacity: isRevealed ? 1 : 0,
            duration: const Duration(milliseconds: 400),
            child: AnimatedSlide(
              offset: isRevealed ? Offset.zero : const Offset(0, 0.2),
              duration: const Duration(milliseconds: 400),
              child: Padding(
                padding: const EdgeInsets.only(bottom: 14),
                child: _dedicationHabitTile(habit),
              ),
            ),
          );
        }),
        if (habits.isNotEmpty) ...[
          AnimatedOpacity(
            opacity: _showVerse ? 1 : 0,
            duration: const Duration(milliseconds: 500),
            child: _milestonePreviewSection(habits),
          ),
          const SizedBox(height: 20),
        ],
        AnimatedOpacity(
          opacity: _showVerse ? 1 : 0,
          duration: const Duration(milliseconds: 500),
          child: Column(
            children: [
              Text(
                '\u201CThe steadfast love of the Lord never ceases; his mercies never come to an end; they are new every morning.\u201D',
                textAlign: TextAlign.center,
                style: TextStyle(
                    fontSize: 13,
                    fontStyle: FontStyle.italic,
                    color: TributeColor.softGold.withValues(alpha: 0.6),
                    height: 1.6),
              ),
              const SizedBox(height: 6),
              Text('Lamentations 3:22\u201323',
                  style: TextStyle(
                      fontSize: 11, color: TributeColor.golden.withValues(alpha: 0.5))),
            ],
          ),
        ),
      ],
    );
  }

  Widget _postDedicationContent() {
    return Column(
      children: [
        const SizedBox(height: 60),
        AnimatedOpacity(
          opacity: _showDedicatedMessage ? 1 : 0,
          duration: const Duration(milliseconds: 800),
          child: AnimatedScale(
            scale: _showDedicatedMessage ? 1 : 0.92,
            duration: const Duration(milliseconds: 800),
            child: Column(
              children: [
                Container(
                  width: 90,
                  height: 90,
                  decoration: BoxDecoration(
                    shape: BoxShape.circle,
                    gradient: RadialGradient(
                      colors: [
                        TributeColor.golden.withValues(alpha: 0.4),
                        TributeColor.golden.withValues(alpha: 0.1),
                      ],
                    ),
                  ),
                  child: const Icon(Icons.auto_awesome, size: 38, color: TributeColor.golden),
                ),
                const SizedBox(height: 24),
                const Text('Your week is dedicated.',
                    style: TextStyle(
                        fontSize: 20,
                        fontWeight: FontWeight.w700,
                        color: TributeColor.warmWhite)),
                const SizedBox(height: 8),
                const Text(
                  'God is with you in the effort and in the rest.',
                  textAlign: TextAlign.center,
                  style: TextStyle(fontSize: 15, color: TributeColor.softGold),
                ),
              ],
            ),
          ),
        ),
      ],
    );
  }

  Widget _dedicationHabitTile(Habit habit) {
    final accent =
        habit.habitTrackingType == HabitTrackingType.abstain ? TributeColor.sage : TributeColor.golden;
    final verse = ScriptureLibrary.anchorVerse(habit.habitCategory);
    final summary = widget.weekCycleManager.weekProjectionSummary(habit);

    return Container(
      padding: const EdgeInsets.all(16),
      decoration: TributeDecorations.card,
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Row(
            children: [
              Container(
                width: 48,
                height: 48,
                decoration: BoxDecoration(
                  shape: BoxShape.circle,
                  gradient: RadialGradient(
                    colors: [
                      accent.withValues(alpha: 0.25),
                      accent.withValues(alpha: 0.06),
                    ],
                  ),
                ),
                child: Icon(_habitIcon(habit), size: 22, color: accent),
              ),
              const SizedBox(width: 12),
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(habit.name,
                        style: const TextStyle(
                            fontSize: 15,
                            fontWeight: FontWeight.w600,
                            color: TributeColor.warmWhite)),
                    Text(habit.purposeStatement,
                        maxLines: 2,
                        overflow: TextOverflow.ellipsis,
                        style: TextStyle(
                            fontSize: 12, color: Colors.white.withValues(alpha: 0.45))),
                  ],
                ),
              ),
              Text(summary,
                  style: TextStyle(
                      fontSize: 10,
                      fontWeight: FontWeight.w500,
                      color: TributeColor.softGold.withValues(alpha: 0.6))),
            ],
          ),
          if (habit.isCompletedToday) ...[
            const SizedBox(height: 10),
            Row(
              children: [
                Icon(Icons.check_circle_rounded, size: 13, color: TributeColor.golden),
                const SizedBox(width: 6),
                Text('Already completed today',
                    style: const TextStyle(fontSize: 12, color: TributeColor.sage)),
              ],
            ),
          ],
          const SizedBox(height: 10),
          Text(
            '\u201C${verse.text}\u201D \u2014 ${verse.reference}',
            maxLines: 2,
            overflow: TextOverflow.ellipsis,
            style: TextStyle(
                fontSize: 12,
                fontStyle: FontStyle.italic,
                color: TributeColor.softGold.withValues(alpha: 0.5)),
          ),
        ],
      ),
    );
  }

  Widget _milestonePreviewSection(List<Habit> habits) {
    final previews = habits
        .map((h) {
          final p = widget.weekCycleManager.microMilestonePreview(h);
          return p != null ? (h, p) : null;
        })
        .whereType<(Habit, String)>()
        .toList();

    if (previews.isEmpty) return const SizedBox.shrink();

    return Container(
      padding: const EdgeInsets.all(16),
      decoration: BoxDecoration(
        color: TributeColor.golden.withValues(alpha: 0.04),
        borderRadius: BorderRadius.circular(14),
        border: Border.all(color: TributeColor.golden.withValues(alpha: 0.12), width: 0.5),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          const Text('If you hit your targets this week:',
              style: TextStyle(
                  fontSize: 14,
                  fontWeight: FontWeight.w600,
                  color: TributeColor.softGold)),
          const SizedBox(height: 12),
          ...previews.map((item) {
            final (habit, preview) = item;
            return Padding(
              padding: const EdgeInsets.only(bottom: 8),
              child: Row(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  const Icon(Icons.auto_awesome, size: 12, color: TributeColor.golden),
                  const SizedBox(width: 8),
                  Expanded(
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Text(habit.name,
                            style: const TextStyle(
                                fontSize: 12,
                                fontWeight: FontWeight.w600,
                                color: TributeColor.warmWhite)),
                        Text(preview,
                            style: TextStyle(
                                fontSize: 11,
                                color: TributeColor.softGold.withValues(alpha: 0.7))),
                      ],
                    ),
                  ),
                ],
              ),
            );
          }),
        ],
      ),
    );
  }

  IconData _habitIcon(Habit habit) {
    if (habit.habitTrackingType == HabitTrackingType.abstain) return Icons.shield_rounded;
    switch (habit.habitCategory) {
      case HabitCategory.gratitude: return Icons.auto_awesome;
      case HabitCategory.scripture: return Icons.menu_book;
      case HabitCategory.exercise: return Icons.fitness_center;
      case HabitCategory.rest: return Icons.bedtime;
      case HabitCategory.fasting: return Icons.no_food;
      case HabitCategory.study: return Icons.school;
      case HabitCategory.service: return Icons.volunteer_activism;
      case HabitCategory.connection: return Icons.people;
      case HabitCategory.health: return Icons.favorite;
      case HabitCategory.abstain: return Icons.shield_rounded;
      case HabitCategory.custom: return Icons.star;
    }
  }
}

File: swift_reference/Views/SundayDedicationView.swift
import SwiftUI
import SwiftData

struct SundayDedicationView: View {
    let weekCycleManager: WeekCycleManager
    let onDedicated: () -> Void

    @Query(sort: \Habit.sortOrder) private var habits: [Habit]

    @State private var showHeading: Bool = false
    @State private var showVerse: Bool = false
    @State private var showButton: Bool = false
    @State private var revealedTileCount: Int = 0
    @State private var isDedicating: Bool = false
    @State private var isDedicated: Bool = false
    @State private var showDedicatedMessage: Bool = false
    @State private var breathe: Bool = false
    @State private var glowIntensity: Double = 0.08

    private var isMidWeekStart: Bool {
        !weekCycleManager.isSunday && !weekCycleManager.isCurrentWeekDedicated
    }

    var body: some View {
        ZStack {
            TributeColor.charcoal.ignoresSafeArea()

            RadialGradient(
                colors: [
                    TributeColor.golden.opacity(glowIntensity),
                    TributeColor.softGold.opacity(glowIntensity * 0.4),
                    Color.clear
                ],
                center: .center,
                startRadius: 0,
                endRadius: breathe ? 320 : 250
            )
            .ignoresSafeArea()
            .animation(.easeInOut(duration: 3).repeatForever(autoreverses: true), value: breathe)

            VStack(spacing: 0) {
                ScrollView {
                    VStack(spacing: 28) {
                        if !isDedicated {
                            preDedicationContent
                        } else {
                            postDedicationContent
                        }
                    }
                    .padding(.horizontal, 24)
                    .padding(.top, 32)
                    .padding(.bottom, 100)
                }

                Spacer(minLength: 0)

                if !isDedicated && showButton && !isDedicating {
                    Button {
                        performDedication()
                    } label: {
                        HStack(spacing: 10) {
                            Image(systemName: "hands.sparkles.fill")
                                .font(.subheadline)
                            Text("Dedicate this week to God")
                        }
                        .tributeButton()
                    }
                    .padding(.horizontal, 24)
                    .padding(.bottom, 32)
                    .transition(.opacity.combined(with: .move(edge: .bottom)))
                }

                if isDedicated && showDedicatedMessage {
                    Button {
                        onDedicated()
                    } label: {
                        HStack(spacing: 8) {
                            Text("Begin your week")
                            Image(systemName: "arrow.right")
                                .font(.subheadline)
                        }
                        .tributeButton()
                    }
                    .padding(.horizontal, 24)
                    .padding(.bottom, 32)
                    .transition(.opacity.combined(with: .move(edge: .bottom)))
                }
            }
        }
        .sensoryFeedback(.success, trigger: isDedicated)
        .onAppear { startEntryAnimations() }
    }

    @ViewBuilder
    private var preDedicationContent: some View {
        VStack(spacing: 6) {
            if isMidWeekStart {
                Text("Starting mid-week?")
                    .font(.system(.caption, design: .serif))
                    .foregroundStyle(TributeColor.softGold.opacity(0.6))
            }

            Text(isMidWeekStart ? "No problem. Let's dedicate\nwhat's left of this week." : "Set Your Week")
                .font(.system(.title2, design: .serif, weight: .bold))
                .foregroundStyle(.primary)
                .multilineTextAlignment(.center)

            Text("Your habits, your purpose, your offering.")
                .font(.subheadline)
                .foregroundStyle(.secondary)
                .multilineTextAlignment(.center)
        }
        .opacity(showHeading ? 1 : 0)
        .offset(y: showHeading ? 0 : 8)

        VStack(spacing: 14) {
            ForEach(Array(habits.enumerated()), id: \.element.id) { index, habit in
                let isRevealed = index < revealedTileCount
                dedicationHabitTile(habit: habit)
                    .opacity(isRevealed ? 1 : 0)
                    .offset(y: isRevealed ? 0 : 12)
                    .animation(.easeOut(duration: 0.4).delay(Double(index) * 0.2), value: revealedTileCount)
            }
        }

        if !habits.isEmpty {
            milestonePreviewSection
                .opacity(showVerse ? 1 : 0)
                .offset(y: showVerse ? 0 : 8)
        }

        VStack(spacing: 6) {
            Text("\u{201C}The steadfast love of the Lord never ceases; his mercies never come to an end; they are new every morning.\u{201D}")
                .font(.system(.subheadline, design: .serif))
                .italic()
                .foregroundStyle(TributeColor.softGold.opacity(0.6))
                .multilineTextAlignment(.center)
            Text("Lamentations 3:22\u{2013}23")
                .font(.caption)
                .foregroundStyle(TributeColor.golden.opacity(0.5))
        }
        .opacity(showVerse ? 1 : 0)
    }

    @ViewBuilder
    private var postDedicationContent: some View {
        Spacer().frame(height: 60)

        VStack(spacing: 24) {
            ZStack {
                Circle()
                    .fill(
                        RadialGradient(
                            colors: [
                                TributeColor.golden.opacity(0.4),
                                TributeColor.golden.opacity(0.1)
                            ],
                            center: .center,
                            startRadius: 0,
                            endRadius: 50
                        )
                    )
                    .frame(width: 90, height: 90)

                Image(systemName: "hands.sparkles.fill")
                    .font(.system(size: 38))
                    .foregroundStyle(TributeColor.golden)
            }

            VStack(spacing: 8) {
                Text("Your week is dedicated.")
                    .font(.system(.title3, design: .serif, weight: .bold))
                    .foregroundStyle(.primary)

                Text("God is with you in the effort and in the rest.")
                    .font(.system(.subheadline, design: .serif))
                    .foregroundStyle(TributeColor.softGold)
                    .multilineTextAlignment(.center)
            }
        }
        .opacity(showDedicatedMessage ? 1 : 0)
        .scaleEffect(showDedicatedMessage ? 1 : 0.92)
    }

    private func dedicationHabitTile(habit: Habit) -> some View {
        VStack(alignment: .leading, spacing: 12) {
            HStack(spacing: 12) {
                ZStack {
                    Circle()
                        .fill(
                            RadialGradient(
                                colors: [
                                    tileAccent(habit).opacity(0.25),
                                    tileAccent(habit).opacity(0.06)
                                ],
                                center: .center,
                                startRadius: 0,
                                endRadius: 24
                            )
                        )
                        .frame(width: 48, height: 48)

                    Image(systemName: habit.habitCategory.iconName)
                        .font(.system(size: 20))
                        .foregroundStyle(tileAccent(habit))
                }

                VStack(alignment: .leading, spacing: 3) {
                    Text(habit.name)
                        .font(.system(.headline, design: .serif))

                    Text(habit.purposeStatement)
                        .font(.caption)
                        .foregroundStyle(.secondary)
                        .lineLimit(2)
                }

                Spacer()

                Text(weekCycleManager.weekProjectionSummary(for: habit))
                    .font(.caption2.weight(.medium))
                    .foregroundStyle(TributeColor.softGold.opacity(0.6))
            }

            if habit.isCompletedToday() && habit.isBuiltIn {
                HStack(spacing: 6) {
                    Image(systemName: "checkmark.circle.fill")
                        .font(.caption)
                        .foregroundStyle(TributeColor.golden)
                    Text("Already completed today")
                        .font(.caption)
                        .foregroundStyle(TributeColor.sage)
                }
            }

            let verse = ScriptureLibrary.anchorVerse(for: habit.habitCategory)
            Text("\u{201C}\(verse.text)\u{201D} \u{2014} \(verse.reference)")
                .font(.caption)
                .italic()
                .foregroundStyle(TributeColor.softGold.opacity(0.5))
                .lineLimit(2)
        }
        .tributeCard()
    }

    @ViewBuilder
    private var milestonePreviewSection: some View {
        let previews = habits.compactMap { habit -> (Habit, String)? in
            guard let preview = weekCycleManager.microMilestonePreview(for: habit) else { return nil }
            return (habit, preview)
        }

        if !previews.isEmpty {
            VStack(alignment: .leading, spacing: 12) {
                Text("If you hit your targets this week:")
                    .font(.system(.subheadline, design: .serif, weight: .semibold))
                    .foregroundStyle(TributeColor.softGold)

                ForEach(previews, id: \.0.id) { habit, preview in
                    HStack(alignment: .top, spacing: 10) {
                        Image(systemName: "sparkle")
                            .font(.caption)
                            .foregroundStyle(TributeColor.golden)
                            .padding(.top, 2)

                        VStack(alignment: .leading, spacing: 2) {
                            Text(habit.name)
                                .font(.caption.weight(.semibold))
                                .foregroundStyle(.primary)
                            Text(preview)
                                .font(.caption)
                                .foregroundStyle(TributeColor.softGold.opacity(0.7))
                        }
                    }
                }
            }
            .padding(16)
            .frame(maxWidth: .infinity, alignment: .leading)
            .background(TributeColor.golden.opacity(0.04))
            .clipShape(.rect(cornerRadius: 14))
            .overlay(
                RoundedRectangle(cornerRadius: 14)
                    .strokeBorder(TributeColor.golden.opacity(0.12), lineWidth: 0.5)
            )
        }
    }

    private func tileAccent(_ habit: Habit) -> Color {
        habit.habitTrackingType == .abstain ? TributeColor.sage : TributeColor.golden
    }

    private func startEntryAnimations() {
        breathe = true

        withAnimation(.easeOut(duration: 0.6).delay(0.3)) {
            showHeading = true
        }

        let tileDelay = 0.8
        for i in 0...habits.count {
            let delay = tileDelay + Double(i) * 0.2
            withAnimation(.easeOut(duration: 0.4).delay(delay)) {
                revealedTileCount = i + 1
            }
        }

        let verseDelay = tileDelay + Double(habits.count) * 0.2 + 0.3
        withAnimation(.easeOut(duration: 0.5).delay(verseDelay)) {
            showVerse = true
        }
        withAnimation(.easeOut(duration: 0.4).delay(verseDelay + 0.3)) {
            showButton = true
        }
    }

    private func performDedication() {
        isDedicating = true

        withAnimation(.easeInOut(duration: 1.5)) {
            glowIntensity = 0.3
        }

        Task {
            try? await Task.sleep(for: .seconds(1.0))

            weekCycleManager.dedicateCurrentWeek()

            withAnimation(.easeInOut(duration: 0.8)) {
                isDedicated = true
                showButton = false
            }

            withAnimation(.easeInOut(duration: 1.0)) {
                glowIntensity = 0.12
            }

            try? await Task.sleep(for: .seconds(0.5))

            withAnimation(.spring(response: 0.6, dampingFraction: 0.8)) {
                showDedicatedMessage = true
            }
        }
    }
}
